equistore-torch/src/labels.cpp

```diff
@@ -293,6 +293,40 @@ TorchLabels LabelsHolder::insert(int64_t index, std::string name, torch::Tensor
 }
 
 
+TorchLabels LabelsHolder::permute(std::vector<int64_t> dims) {
+    auto names = this->names();
+
+    if (dims.size() != names.size()) {
+        C10_THROW_ERROR(
+            ValueError,
+            "number of dimensions Labels (" + std::to_string(names.size()) + ") does " +
+            "not match the length of the desired ordering of dimensions (" +
+            std::to_string(dims.size()) + ")."
+        );
+    }
+
+    auto new_names = std::vector<std::string>();
+
+    for (auto index : dims) {
+        if (index < 0) {
+            index += names.size();
+        }
+        new_names.push_back(names.at(index));
+    }
+
+    if (std::set<std::string>(new_names.begin(), new_names.end()).size() != names.size()) {
+        C10_THROW_ERROR(
+            ValueError,
+            "duplicate dims are not allowed"
```

Comment:
```suggestion
            "duplicate dimensions are not allowed"
```

Also, should we do this check on the integer values instead, after normalizing `-1` to `size -1`?